.github/workflows/CICD.yml

```diff
@@ -1,18 +1,148 @@
 name: CICD
+on: [pull_request, push]
+jobs:
+  test-lib:
+    name: Test library
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        toolchain: [stable, nightly]
+    continue-on-error: true
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Rust
+        run: |
+          rustup toolchain add ${{ matrix.toolchain }} --component llvm-tools-preview
+          rustup override set ${{ matrix.toolchain }}
+      - name: Configure cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: test-${{ runner.os }}-cargo-${{ matrix.toolchain }}-${{ hashFiles('**/Cargo.lock') }}
+      - name: Test
+        run: cargo test --lib
+  test-coverage:
+    name: Test coverage (gcc-${{ matrix.gcc }}, clang-${{ matrix.clang }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - gcc: 5
```

Comment:
@marco-c I think we could remove gcc 5-6, gcc version in ubuntu 18.04 is 7.4.

Reply:
In addition to that, should I add newer combinations like clang 12 and 13 as well?

Reply:
Yes, see https://github.com/mozilla/grcov/pull/667#issuecomment-985666987.